generators/app/templates/gulp/tasks/vendor.js

```diff
@@ -0,0 +1,51 @@
+'use strict';
+
+var fs = require('fs');
+var path = require('path');
+
+var bundleVendorTask = function (gulp, plugins, config) {
+  var fileListPath = path.join(config.src.base, 'scripts/vendor.json');
+  var getFileList = function () {
+    return JSON.parse( fs.readFileSync(fileListPath) );
+  };
+
+  gulp.task('vendor-clean', function(done) {
+    return plugins.del([ path.join(config.dest.base, 'scripts/vendor.js') ], done);
+  });
+
+  gulp.task('vendor-watch', function() {
+    var fileList = getFileList();
+
+    if (fileList.length) {
+      return gulp.src(fileList)
+        .pipe(plugins.concat('vendor.js'))
+        .pipe(gulp.dest(path.join(config.dest.base, config.dest.scripts)))
+        .pipe(plugins.browserSync.stream({ match: '**/*.js' }));
+    }
+
+    <% if(projectType == 'wp-with-fe') { %>
+    return gulp.start('vendor-clean', function () {
```

Comment:
It's connected with the issue above `Please verify mix of return and callback`. Let's resolve it and go back to this code chunk later. Might be possible to get rid of this part entirely!